### PR TITLE
[FEATURE] Export annotated spectra from Sirius as mzml

### PR DESCRIPTION
**Pull request by jpfeuffer**
# Description

To be able to visualize matched spectra. The plan is to do a lookup by spectra_id between the original spectra and
the ones exported here.
Exporting multiple possible annotations in IDs of a IdXML did not seem like a good solution.

# Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

## How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

</details>

## Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by timosachsenberg** [comment]
> Exporting multiple possible annotations in IDs of a IdXML did not seem like a good solution.

why again? Afaik this would be already supported in TOPPView

**Comment by jpfeuffer** [comment]
How? I don't even know how to do an idXML for metabo-only. Needs to work for featureXML, too. Annotations would be multiplied as well since in featureXMLs you may have multiple MS2 spectra that go into Sirius and you get one annotation out. Apparently, it contains all fragments which matched at least one peak in one of the MS2 spectra or something like that.

**Comment by timosachsenberg** [comment]
Do you consider also exporting MSP files? Human readable and maybe usable for libraries?

**Comment by jpfeuffer** [comment]
They are already msp files basically. That is what I am reading in

**Comment by timosachsenberg** [comment]
k sounds good. is this so general that we could have a msp to MSExperiment/mzML converter?

**Comment by jpfeuffer** [comment]
No, it is a full Sirius workspace that is needed, to associate it to the correct compounds etc.
We can make a Sirius Workspace reader but I don't think it is worth it.

File: src/openms/source/FORMAT/DATAACCESS/SiriusFragmentAnnotation.cpp [hunk 1]
```diff
@@ -240,8 +281,125 @@ namespace OpenMS
...
+          MSSpectrum::StringDataArray fragments_ionization;
+          if (use_exact_mass)
+          {
+            fragments_exactmasses.setName("mz");
+          }
+          else
+          {
+            fragments_exactmasses.setName("exact_mass");
+          }
+          fragments_explanations.setName("explanation");
```
[the remark this answers is not in the document]
Comment (jpfeuffer) [reply src/openms/source/FORMAT/DATAACCESS/SiriusFragmentAnnotation.cpp:422]: But then it needs to be clarified and defined, how ionnames and charges need to look like.

[the remark this answers is not in the document]
Comment (timosachsenberg) [reply src/openms/source/FORMAT/DATAACCESS/SiriusFragmentAnnotation.cpp:422]: For charges I prefer integer data arrays if it makes sense here. No strong feeling about the ion names but having a meta value name would be a first step to make this more consistent. Or make it like SIRIUS_IONNAME

**Review by timosachsenberg — APPROVED** [review]
thanks for taking the time to update this

**Comment by jpfeuffer** [comment]
Sorry, but I am not going to debug this. I will disable the test on linux.

**Comment by timosachsenberg** [comment]
they have basically same mass so this is probably just non-deterministic :( . feel free to disable

**Pull request merged**